lib/honey/translator.ex

```diff
@@ -107,77 +87,71 @@ defmodule Honey.Translator do
       :bpf_printk ->
         [[string | other_params]] = params
 
-        if(!is_bitstring(string)) do
-          throw(
-            "First argument of bpf_printk must be a string. Received: #{Macro.to_string(params)}"
-          )
+        if !is_bitstring(string) do
+          raise "First argument of bpf_printk must be a string. Received: #{Macro.to_string(params)}"
         end
 
         string = String.replace(string, "\n", "\\n")
+        code_vars = Enum.map(other_params, & to_c(&1, context))
 
-        code_vars =
-          Enum.map(other_params, fn expr ->
-            to_c(expr, context)
-          end)
-
-        code =
-          Enum.reduce(code_vars, "", fn translated, so_far ->
-            so_far <> translated.code
-          end)
+        code = Enum.reduce(code_vars, "", fn %{code: code}, so_far -> so_far <> code end)
 
         vars =
           Enum.reduce(code_vars, "", fn translated, so_far ->
             so_far = if so_far != "", do: so_far <> ", ", else: ""
             so_far <> translated.return_var_name <> ".value.integer"
           end)
 
-        result_var = get_new_helper_var_name()
-
-        code =
-          code <>
-            "bpf_printk(\"#{string}\", #{vars});\n" <>
-            "Generic #{result_var} = {.type = INTEGER, .value.integer = 0};"
+        result_var = unique_helper_var()
 
         # TODO: Return nil insted of 0, as Elixir would
-
-        Honey.TranslatedCode.new(code, result_var)
+        # huh?
```

Comment:
Oh, I thought _bpf_printk_ returned _void_. In this case, an Elixir equivalent could be _nil_, and that's what I meant.
But since _bpf_printk_ returns a _long_, we should reflect it in the code.